solt/core/_base_transforms.py

```diff
@@ -541,8 +539,13 @@ def _apply_img(self, img: np.ndarray, settings: dict):
             Output Image
 
         """
+        ensure_valid_image(img, num_dims=(2,))
```

Comment:
This is done in a wrong place. The decorator has been checking that the output of the method returns a 3-dim shape.